pkg/binaries/run/run_test.go

```diff
@@ -214,3 +212,35 @@ func fakeExecCommand(testName string) run.CmdFn {
 		return cmd
 	}
 }
+
+const (
+	testStdout = "this is stdout\n"
+	testStderr = "this is stderr\n"
+)
+
+func TestRunCmdStdoutAndStderr(t *testing.T) {
+	t.Run("Should get both stdout and stderr", func(t *testing.T) {
+		cmdFn := fakeExecCommand("StdoutAndStderr")
+		var out bytes.Buffer
+
+		outResult, err := run.New(nil, "working_dir", "binary_path", []string{}, cmdFn).
+			Run(&out, []string{})
+		assert.NoError(t, err)
+
+		expectedOutput := testStdout + testStderr
+
+		assert.Equal(t, expectedOutput, string(outResult))
+		assert.Equal(t, expectedOutput, out.String())
+	})
+}
+
+func TestRunProcessStdoutAndStderr(t *testing.T) {
```

Comment:
Can you explain this more? 

I've never seen an `os.Exit()` in a test before.

Reply:
Why does it care about GO_TEST_PROCESS?

and does it really test anything? It just prints out some text?

Reply:
I'm following the pattern from `TestRunProcessSuccess`. This main use of this function is not to be directly, but by running `go -test.run ...", see https://github.com/oslokommune/okctl/blob/c754d59d85b481da47d31b532a23e76974852d99/pkg/binaries/run/run_test.go#L199

So the flow is:
* The start point is `TestRunCmdStdoutAndStderr`
* In that test, we call `fakeExecCommand("StdoutAndStderr")` which creates a cmd function that executes `go -test.run=TestRunProcessStdoutAndStderr`.
* By calling cmd.Exec, that function is invoked, and it prints stuff to stdout and stderr.
* We collect the output, and do asserts.

GO_TEST_PROCESS is used to ignore the test if run directly when running tests in okctl. GO_TEST_PROCESS is set to 1 in the code flow described above, so we're able to distinguish when the function is called directly, or via the flow above.

Another implementation of all this would be to create a shell file or binary and push it to git (like we do for the upgrade tests, for instance). I did not consider this, I just followed the existing pattern. For code readability, we can consider this.

Reply:
I'm following the pattern from `TestRunProcessSuccess`. This main use of this function is not to be directly, but by running `go -test.run ...", see https://github.com/oslokommune/okctl/blob/c754d59d85b481da47d31b532a23e76974852d99/pkg/binaries/run/run_test.go#L199

So the flow is:
* The start point is `TestRunCmdStdoutAndStderr`
* In that test, we call `fakeExecCommand("StdoutAndStderr")` which creates a cmd function that executes `go -test.run=TestRunProcessStdoutAndStderr`.
* By calling cmd.Exec, that function is invoked, and it prints stuff to stdout and stderr.
* We collect the output, and do asserts.

GO_TEST_PROCESS is used to ignore the test if run directly when running tests in okctl. GO_TEST_PROCESS is set to 1 in the code flow described above, so we're able to distinguish when the function is called directly, or via the flow above.

Another implementation of all this would be to create a shell file or binary and push it to git (like we do for the upgrade tests, for instance). I did not consider this, I just followed the existing pattern. For code readability, we can consider this.

Reply:
If there is no `os.Exit(0)`, the test fails with this:

```
        	            	expected: "this is stdout\nthis is stderr\n"
        	            	actual  : "this is stdout\nthis is stderr\nPASS\n"
```

I have no idea what the "PASS" is.

Reply:
Ahhh right. That explains it. Thanks